File: python/Calculator.py
def calculate(num1, operator, num2):
    if operator == '+':
        return num1 + num2
    elif operator == '-':
        return num1 - num2
    elif operator == '*':
        return num1 * num2
    elif operator == '/':
        if num2 != 0:
            return num1 / num2
        else:
            return "错误：除数不能为零"
    else:
        return "无效的操作符"

print("简单计算器")
print("可用操作: +, -, *, /")
print("输入 'q' 退出")

while True:
    # 获取第一个数
    num1 = input("请输入第一个数: ")
    if num1.lower() == 'q':
        break

    try:
        num1 = float(num1)
    except ValueError:
        print("无效的输入，请输入一个数字")
        continue

    # 获取操作符
    operator = input("请输入操作符 (+, -, *, /): ")
    if operator.lower() == 'q':
        break

    if operator not in ['+', '-', '*', '/']:
        print("无效的操作符")
        continue

    # 获取第二个数
    num2 = input("请输入第二个数: ")
    if num2.lower() == 'q':
        break

    try:
        num2 = float(num2)
    except ValueError:
        print("无效的输入，请输入一个数字")
        continue

    # 计算并显示结果
    result = calculate(num1, operator, num2)
    if isinstance(result, str):
        print(result)
    else:
        print(f"结果: {num1} {operator} {num2} = {result:.2f}")

    print()  # 打印空行以增加可读性

print("感谢使用！")

File: src/com/aixcoder/java/Calculator.java
package com.aixcoder.java;

import java.util.Scanner;

public class Calculator {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        double num1, num2, result;
        char operator;

        while (true) {
            System.out.println("简单计算器");
            System.out.println("可用操作: +, -, *, /");
            System.out.println("输入 'q' 退出");

            System.out.print("请输入第一个数: ");
            if (scanner.hasNext("q")) {
                break;
            }
            num1 = scanner.nextDouble();

            System.out.print("请输入操作符 (+, -, *, /): ");
            operator = scanner.next().charAt(0);

            System.out.print("请输入第二个数: ");
            num2 = scanner.nextDouble();

            switch (operator) {
                case '+':
                    result = num1 + num2;
                    break;
                case '-':
                    result = num1 - num2;
                    result = num1 - num2;
                    break;
                case '*':
                    result = num1 * num2;
                    break;
                case '/':
                    if (num2 != 0) {
                        result = num1 / num2;
                    } else {
                        System.out.println("错误：除数不能为零");
                        continue;
                    }
                    break;
                default:
                    System.out.println("无效的操作符");
                    continue;
            }

            System.out.printf("结果: %.2f %c %.2f = %.2f\n", num1, operator, num2, result);
            System.out.println();
        }

        System.out.println("感谢使用！");
        scanner.close();
    }
}